api/src/lib.rs

```diff
@@ -84,7 +84,15 @@ pub async fn submit_request(input: HttpRequest) -> Result<Value, Box<dyn Error>>
         HttpMethod::HEAD => todo!(),
     };
 
+    let res_type = res.headers().get("content-type").unwrap().to_str().unwrap();
+
+    if !res_type.starts_with("application/json") {
+        // I couldn't figure out how to safely throw an error so I'm just returning this for now
+        println!("expected application/json, got {}", res_type);
+        return Ok(json!({"msg": "not JSON!"}));
+    }
+
     let res_str = res.text().await?;
-    let res_json = serde_json::from_str(&res_str).unwrap();
+    let res_json = serde_json::from_str(&res_str).unwrap_or_default();
     Ok(res_json)
```

Comment:
one thing I am wondering, do we want to stick with only JSON being the valid return type for now or do we want to bake in support for non-json responses like HTML or text/binary?

fine if we go with the former approach but just wanted to make sure it was in line with your vision

Reply:
I wanted to bake in support for other response types, I just defaulted to json to make something work for now. I think that once we get header support in then we can use the response type header to have a `match` statement in order to call the right deserialization method before returning to the ui.

I think that would necessitate the creation of something like a Response wrapper so the `submit_request` could still maintain a single output type? Open to some thoughts on to best handle that though.